src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -1730,6 +1730,31 @@ public static StatementSyntax ParseStatement(string text, int offset = 0, ParseO
             }
         }
 
+        /// <summary>
+        /// Parse a MemberDeclarationSyntax. This includes all of the kinds of members that could occur in a struct declaration.
+        /// If nothing resembling a valid member declaration is found in the input, returns null.
+        /// </summary>
+        /// <param name="text">The text of the declaration.</param>
+        /// <param name="offset">Optional offset into text.</param>
+        /// <param name="options">The optional parse options to use. If no options are specified default options are
+        /// used.</param>
+        /// <param name="consumeFullText">True if extra tokens in the input following a declaration should be treated as an error</param>
+        public static MemberDeclarationSyntax ParseMemberDeclaration(string text, int offset = 0, ParseOptions options = null, bool consumeFullText = true)
+        {
+            using (var lexer = MakeLexer(text, offset, (CSharpParseOptions)options))
+            using (var parser = MakeParser(lexer))
+            {
+                var node = parser.ParseMemberDeclaration();
+                if (node == null)
+                {
+                    return null;
+                }
+
+                if (consumeFullText) node = parser.ConsumeUnexpectedTokens(node);
```

Comment:
missing braces.

Reply:
This code appears identically many places in this source file. [When in Rome](https://en.wikipedia.org/wiki/When_in_Rome,_do_as_the_Romans_do)...

Reply:
This is a minor adjustment to an existing source file, where this line appears verbatim 10 times as written here, and zero times as suggested. The "coding guidelines" you're suggesting to apply here are not followed by the compiler in general, and certainly not in this source file, and I do not intend to differ from the local style. It would be distractingly different to the reader to change it here. Also in this very method you'll see a using statement without a block body, and three locals declared with var without the type appearing on the right-hand-side. That is the local style and it should be followed.

---
In reply to: [210342218](https://github.com/dotnet/roslyn/pull/29263#discussion_r210342218) [](ancestors = 210342218,210049253)

Reply:
The [coding guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) for `if` do allow for omission of braces for the single line case. But it does require the statement be on a separate line and properly indented.